Sources/StructuralCore/Structural.swift

```diff
@@ -21,103 +74,187 @@ public protocol Structural {
     public init(_ rawValue: RawValue, _ associatedValues: AssociatedValues) {
         self.name = ""
         self.rawValue = rawValue
         self.associatedValues = associatedValues
     }
 
-    public init(_ name: StaticString, _ rawValue: RawValue, _ associatedValues: AssociatedValues) {
+    /// Creates a representation from the given raw value and associated values.
+    public init(
+        _ name: String, _ rawValue: RawValue, _ associatedValues: AssociatedValues
+    ) {
         self.name = name
         self.rawValue = rawValue
         self.associatedValues = associatedValues
     }
 }
 
-/// Structural representation of a heterogeneous list cons cell.
+/// The structural representation of a chain of properties or associated values.
+///
+/// `StructuralCons<...>` is the compiler-synthesized
+/// `StructuralRepresentation` for chaining struct stored properties and enum
+/// case associated values.
+///
+/// - Parameter Value: the type of the first value.
+/// - Parameter Next: the type of the second value.
+///
+/// - Note: `StructuralCons<...>` is used as a type-level heterogeneous cons
+///   list, which is verbose and not particularly human-readable. If variadiac
+///   generics are added to the Swift language, `StructuralCons` can be
+///   deprecated in favor of a less-verbose heterogeneous list representaiton
+///   using variadic generics.
 public struct StructuralCons<Value, Next> {
+    /// The first value.
     public var value: Value
+    /// The next value.
     public var next: Next
 
+    /// Creates a representation from the given values.
     public init(_ value: Value, _ next: Next) {
         self.value = value
         self.next = next
     }
 }
 
-/// Structural representation of an empty heterogeneous list cons cell.
+/// The structural representation of an empty list.
+///
+/// - Note: `StructuralEmpty<...>` is used as the base case of the type-level
+///   heterogeneous cons list `StructuralCons`. If variadiac generics are added
+///   to the Swift language, `StructuralCons` can be deprecated in favor of a
+///   less-verbose heterogeneous list representation using variadic generics.
 public struct StructuralEmpty {
     public init() {}
 }
 
-/// Structural representation of an alternative between either Left or Right.
+/// The structural representation of an alternative between either `Left` or
+/// `Right`.
+///
+/// `StructuralEither<...>` is the compiler-synthesized
+/// `StructuralRepresentation` for chaining together enum cases.
+///
+/// - Parameter Left: the type of the first alternative.
+/// - Parameter Right: the type of the second alternative.
+///
+/// - Note: `StructuralEither<...>` is used as a type-level heterogeneous cons
+///   list, which is verbose and not particularly human-readable. If variadiac
+///   generics are added to the Swift language, `StructuralEither` can be
+///   deprecated in favor of a less-verbose heterogeneous list representaiton
+///   using variadic generics.
```

Comment:
Same comment regarding variadic generics as above.

Reply:
Removed for now.